napari/layers/utils/layer_utils.py

```diff
@@ -204,71 +214,83 @@ def _nanmax(array):
+                "RGB dtype is not np.uint8. Vispy will cast to np.uint8",
+                deferred=True,
+            )
+        )
+    if not np.issubdtype(dtype, np.integer) and chunk_size:
+        return None
+
+    if chunk_size:
+        shape = _get_blocks_grid_shape(data.shape, chunk_size)
+
+    if data.size > PIXEL_THRESHOLD and (data.ndim == 1):
+        slices = _get_1d_slices(shape, chunk_size)
         reduced_data = [
             [_nanmax(data[sl]) for sl in slices],
             [_nanmin(data[sl]) for sl in slices],
         ]
-    elif data.size > 1e7:
-        # If data is very large take the average of the top, bottom, and
-        # middle slices
+        if not reduced_data:
+            return None
+        if chunk_size:
+            reduced_data = dask.compute(*reduced_data)
+    elif data.size > PIXEL_THRESHOLD:
+        # If data is very large take the top, bottom, and middle slices
         offset = 2 + int(rgb)
-        bottom_plane_idx = (0,) * (data.ndim - offset)
-        middle_plane_idx = tuple(s // 2 for s in data.shape[:-offset])
-        top_plane_idx = tuple(s - 1 for s in data.shape[:-offset])
-        idxs = [bottom_plane_idx, middle_plane_idx, top_plane_idx]
-        # If each plane is also very large, look only at a subset of the image
-        if (
-            np.prod(data.shape[-offset:]) > 1e7
-            and data.shape[-offset] > 64
-            and data.shape[-offset + 1] > 64
-        ):
-            # Find a central patch of the image to take
-            center = [int(s // 2) for s in data.shape[-offset:]]
-            central_slice = tuple(slice(c - 31, c + 31) for c in center[:2])
-            reduced_data = [
-                [_nanmax(data[idx + central_slice]) for idx in idxs],
-                [_nanmin(data[idx + central_slice]) for idx in idxs],
-            ]
+        # Indices are either numpy array or chunk indices dependent on data structure of data and are 0 < length <= 3
+        idxs = _get_plane_indices(shape, offset)
+
+        if chunk_size:
+            slices = _get_crop_slices(shape, idxs, offset, chunk_size)
+            if slices:
+                reduced_data = [
+                    [_nanmax(data[sl]) for sl in slices],
+                    [_nanmin(data[sl]) for sl in slices],
+                ]
+                reduced_data = dask.compute(*reduced_data)
+            else:
+                # this is none when the chunk size product goes over the pixel threshold of 1e7
+                return None
+        # If we get here we are dealing with a numpy array and shape corresponds to the shape of this array.
         else:
+            slices = _get_crop_slices(shape, idxs, offset)
             reduced_data = [
-                [_nanmax(data[idx]) for idx in idxs],
-                [_nanmin(data[idx]) for idx in idxs],
+                [_nanmax(data[slicer[0], slicer[1]]) for slicer in slices],
+                [_nanmin(data[slicer[0], slicer[1]]) for slicer in slices],
```

Comment:
Why we cannot merge this to not use `if` and generate proper slices in `_get_crop_slices`